lmcache/v1/cache_controller/controllers/kv_controller.py

```diff
@@ -139,14 +139,21 @@ async def deregister(self, instance_id: str, worker_id: int) -> None:
         """
         Deregister all kv chunks of an instance-worker.
         """
-        for key in self.kv_pool:
-            self.kv_pool[key] = [
+        keys_to_delete = []
+        for key, metadata_list in self.kv_pool.items():
+            # Filter in-place by modifying the list
+            metadata_list[:] = [
```

Comment:
nit: can we change the variable name to `filtered_metadata_list`?

```suggestion
            filtered_metadata_list = [
```